node/lib/util/stash_util.js

```diff
@@ -107,6 +117,62 @@ exports.makeLogMessage = co.wrap(function *(repo) {
 WIP on ${branchDesc}: ${GitUtil.shortSha(head.id().tostrS())} ${message}`;
 });
 
+
+function getNewGitModuleSha(diff) {
+    const numDeltas = diff.numDeltas();
+    for (let i = 0;  i < numDeltas; ++i) {
+        const delta = diff.getDelta(i);
+        // We assume that the user hasn't deleted the .gitmodules file.
+        // That would be bonkers.
+        const file = delta.newFile();
+        const path = file.path();
+        if (path === GITMODULES) {
+            return delta.newFile().id();
+        }
+    }
+    // diff does not include .gitmodules
+    return null;
+}
+
+
+const stashGitModules = co.wrap(function *(repo, headTree) {
+    assert.instanceOf(repo, NodeGit.Repository);
+
+    const result = {};
+    // RepoStatus throws away the diff new sha, and rather than hack
+    // it, since it's used all over the codebase, we'll just redo the
+    // diffs for this one file.
+
+    const workdirToTreeDiff =
+          yield NodeGit.Diff.treeToWorkdir(repo,
+                                           headTree,
+                                           {pathspec: [GITMODULES]});
```

Comment:
change since last version: s/paths/pathspec/ -- we used the wrong variable name, leading to slooooow stashes.